logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserMapper.kt

```diff
@@ -2,36 +2,88 @@ package com.wire.kalium.logic.data.user
+import com.wire.kalium.persistence.dao.UserEntity
 
 interface UserMapper {
     fun fromApiModel(selfUserInfoResponse: SelfUserInfoResponse): SelfUser
-    fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): PersistedUser
-    fun fromApiModelToDaoModel(selfUserInfoResponse: SelfUserInfoResponse): PersistedUser
-    fun fromDaoModel(user: PersistedUser): SelfUser
+    fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): UserEntity
+    fun fromApiModelToDaoModel(selfUserInfoResponse: SelfUserInfoResponse): UserEntity
+    fun fromDaoModel(user: UserEntity): SelfUser
+
+    /**
+     * Maps the user data to be updated. if the parameters [newName] [newAccent] [newAssetId] are nulls,
+     * it indicates that not updation should be made.
+     *
+     *  TODO: handle deletion of assets references, emptyAssetList
+     */
+    fun fromModelToUpdateApiModel(user: SelfUser, newName: String?, newAccent: Int?, newAssetId: String?): UserUpdateRequest
 }
 
 internal class UserMapperImpl(private val idMapper: IdMapper) : UserMapper {
 
     override fun fromApiModel(selfUserInfoResponse: SelfUserInfoResponse): SelfUser {
-        return SelfUser(idMapper.fromApiModel(selfUserInfoResponse.qualifiedId))
+        return SelfUser(
+            idMapper.fromApiModel(selfUserInfoResponse.qualifiedId),
+            selfUserInfoResponse.name,
+            selfUserInfoResponse.handle,
+            selfUserInfoResponse.email,
+            selfUserInfoResponse.phone,
+            selfUserInfoResponse.accentId,
+            selfUserInfoResponse.team,
+            emptyList()
+        )
     }
 
-    override fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): PersistedUser {
-        return PersistedUser(idMapper.fromApiToDao(userDetailsResponse.id), userDetailsResponse.name, userDetailsResponse.handle)
+    override fun fromApiModelToDaoModel(userDetailsResponse: UserDetailsResponse): UserEntity {
+        return UserEntity(
+            idMapper.fromApiToDao(userDetailsResponse.id),
+            userDetailsResponse.name,
+            userDetailsResponse.handle,
+            null,
+            null,
+            userDetailsResponse.accentId,
+            null
+        )
     }
 
-    override fun fromDaoModel(user: com.wire.kalium.persistence.dao.User): SelfUser {
-        return SelfUser(idMapper.fromDaoModel(user.id))
+    override fun fromDaoModel(user: com.wire.kalium.persistence.dao.UserEntity) =
+        SelfUser(idMapper.fromDaoModel(user.id), user.name, user.handle, user.email, user.phone, user.accentId, user.team, emptyList())
+
+    override fun fromModelToUpdateApiModel(
+        user: SelfUser,
+        newName: String?,
+        newAccent: Int?,
+        newAssetId: String?
+    ): UserUpdateRequest {
+        return UserUpdateRequest(
+            id = user.id.value,
+            qualifiedId = idMapper.toApiModel(user.id),
+            name = newName ?: user.name,
```

Comment:
I suppose we don't want to reset the name to `null` if the `newName` is `null`?

How is this serialized to JSON? Is name property not included in the JSON object if it's null?

Reply:
exactly that was the idea, but after checking we don't serialize null so removed the logic for that handling